packages/language-server/src/lib/FileSystemProvider.ts

```diff
@@ -0,0 +1,94 @@
+import { stat, readdir, Stats } from 'fs';
+import { promisify } from 'util';
+import {
+    FileStat,
+    FileSystemProvider as CSSFileSystemProvider,
+    FileType
+} from 'vscode-css-languageservice';
+import { urlToPath } from '../utils';
+
+interface StatLike {
+    isDirectory(): boolean;
+    isFile(): boolean;
+    isSymbolicLink(): boolean;
+}
+
+export class FileSystemProvider implements CSSFileSystemProvider {
```

Comment:
You added this to lib and it's only used inside `plugins/css`. What's the reason for this? Do you think this could be useful outside the CSS plugin, too, or that it's not directly connected to the CSS plugin?

Reply:
`vscode-html-languageservice` has the same API for path completion. Originally I was thinking maybe we can add it too. But I am not sure is useful inside a svelte file. Since most of the time, the relative path to an asset does work like plain HTML. I think I'll move it to `plugins/css` for now. We could move it out if we want to use it in HTML later.